### PR TITLE
Some things we came across today.

### PR DESCRIPTION
**Pull request by blaggacao**
For us it only worked with ubuntu with kernel 3.15.4 - else there was this error: http://raspberrypi.stackexchange.com/questions/3394/crosstool-ng-build-fails-on-ubuntu-12-04

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,16 +1,16 @@
-FROM debian:6
+FROM ubuntu:trusty
 MAINTAINER Bilge <bilge@scriptfusion.com>
 
 WORKDIR	/root
 
 # Install dependencies.
 RUN	apt-get update && DEBIAN_FRONTEND=noninteractive\
-	apt-get install -y build-essential gperf bison flex texinfo wget gawk libtool automake libncurses5-dev
+	apt-get install -y build-essential gperf bison flex texinfo wget gawk libtool automake libncurses5-dev help2man
```
Comment (Bilge) [inline Dockerfile:8]: What is `help2man` for? Is it required now?

Comment (blaggacao) [inline Dockerfile:8]: It was a dependency of crosstool 1.22 - we didn´t investigate further on that, we just added it.

**Comment by Bilge** [comment]
Please update `.travis.sh` as well. Parity has to be manually maintained between `Dockerfile` and `.travis.sh` because Travis cannot run Docker itself, so we emulate the same set of instructions in this shell script.

**Comment by blaggacao** [comment]
Builds passed but on our local machine we get another error while "Retrieving gcc-"

Notably, gcc has no version specified. Might be a crosstool bug?

**Comment by Bilge** [comment]
[The same thing happened on Travis](https://travis-ci.org/Docker-nano/crosstool-NG/builds/104912119). The build is just not set up correctly to determine error states, for some reason.

**Comment by blaggacao** [comment]
We are investigating and it seems we also need to update the config file, we identified this variable

```
${CT_CC_GCC_VERSION}
```

on http://crosstool-ng.org/git/crosstool-ng/tree/scripts/build/cc/100-gcc.sh?h=1.22#n24

Which is not set in the original config.

If you have any complementary ideas, be aware, that we are relative newbies in this field, so don't hesitate to add comments and suggestions.

however it get's a step further with this variable set. We'll make some additional tests and try to build everything and update the branch.

**Comment by Bilge** [comment]
I haven't looked at crosstool for over a year myself. If you want an authoritative answer on problems you experience while using it, I recommend you head over to the #crosstool-ng chat room on Freenode.

File: .travis.sh [hunk 2]
```diff
@@ -4,13 +4,13 @@ cd ~
...
-	apt-get install -y build-essential gperf bison flex texinfo wget gawk libtool automake libncurses5-dev
+	apt-get install -y build-essential gperf bison flex texinfo wget gawk libtool automake libncurses5-dev help2man
 
 # Download and compile crosstool-NG.
-wget http://crosstool-ng.org/download/crosstool-ng/crosstool-ng-1.20.0.tar.xz 2>&1 &&\
+wget http://crosstool-ng.org/download/crosstool-ng/crosstool-ng-1.22.0.tar.xz 2>&1 &&\
 	tar xf crosstool-ng-*.tar* &&\
 	(
-		cd crosstool-ng-* &&\
+		cd crosstool-ng* &&\
```
Comment (Bilge) [inline .travis.sh]: Shouldn't you remove the asterisk (`*`) here as well? That would keep it the same as in the `Dockerfile`, even though it probably makes no difference to how well it works.

Comment (blaggacao) [inline .travis.sh]: Absolutely right. Done.

**Comment by blaggacao** [comment]
There is an issue with the new Let's Encrypt Certificates of the wget uclibc. I suppose that the travis host doesn't have the Let's Encrypt authority under accredited authority yet.

We can handle that in the dockerfile, but I wouldn't know how to do that in travis...

**Comment by blaggacao** [comment]
Will have to make a squash force push once we get this working ...

**Comment by Bilge** [comment]
Why do you want to update the crosstool-NG version anyway? Is there something wrong with the toolchain I've already built?

**Comment by Bilge** [comment]
Thanks for these changes. These have been incorporated into the master branch. The `travis` branch will be deleted soon as Travis support has also been added to the master branch.

**Pull request closed without merging**